src/app/store/services/shared.service.ts

```diff
@@ -269,6 +270,63 @@ export function getComposeMailShortcuts(component: ComposeMailComponent) {
   ];
 }
 
+export function getContactsSthortcuts(component: MailContactComponent) {
+  return [
+    getShortcutKeyObj('right', 'Contacts', 'Enter contact details', () => {
+      component.editContact(null, null);
+    }),
+    getShortcutKeyObj('left', 'Contacts', 'Exit contact details', () => {
+      component.destroySplitContactLayout();
+    }),
+    getShortcutKeyObj('t', 'Contacts', 'Delete contact', () => {
+      component.openConfirmDeleteModal(component.confirmDeleteModal);
+    }),
+    getShortcutKeyObj('down', 'Contacts', 'Moving between contacts', () => {
+      const selectedContact = component.selectedContact;
+      const contacts = component.contactsState.contacts;
+      if (contacts.length > 0) {
+        if (!selectedContact) {
+          component.selectedContact = contacts[0];
+          component.editContact(component.selectedContact, component.addUserContent);
+        } else {
+          contacts.forEach((contact, index) => {
+            if (selectedContact.id === contact.id) {
+              if (contacts.length > index + 1) {
+                component.selectedContact = contacts[index + 1];
+              }
+              return;
+            }
+          });
+          component.editContact(component.selectedContact, component.addUserContent);
+        }
+      }
+
+    }),
+    getShortcutKeyObj('up', 'Contacts', 'Moving between contacts', () => {
+      const selectedContact = component.selectedContact;
+      const contacts = component.contactsState.contacts;
```

Comment:
both of these codes are very similiar for up and down ,
you should create a common function for them.

Reply:
there is some minor changes, in both codes, in function repetition could be removed.
Okay, I'll update.